Source/NSubstitute/Routing/RouteFactory.cs

```diff
@@ -67,8 +67,8 @@ public IRoute RecordCallSpecification(ISubstituteState state)
         public IRoute RecordReplay(ISubstituteState state)
         {
             return new Route(RouteType.RecordReplay, new ICallHandler[] {
-                new ClearUnusedCallSpecHandler(state)
```

Comment:
Performance impact is negligible here. But the code lost the knowledge that it should clear specs, that's critical for me. Thank for the change :)